### ISSUE TITLE
`get_errors` fails with invalid unicode error strings

### ISSUE THREAD

**Issue by oditorium** [madmaze/pytesseract#561]
The current code from line 170 reads as follows

```
def get_errors(error_string):
    return ' '.join(
        line for line in error_string.decode(DEFAULT_ENCODING).splitlines()
    ).strip()
```

However, tesseract can sometimes produce invalid Unicode values in `error_string` (eg MacOS Sequioa 15.1.1, see #562 ) in which case this raises an exception and the original error message is lost.

The fix is easy and probably uncontroversial: just add `, errors="replace"` and this fails gracefully

```
def get_errors(error_string):
    return ' '.join(
        line for line in error_string.decode(DEFAULT_ENCODING, errors="replace").splitlines()
    ).strip()
```

**Comment by bozhodimitrov** [comment]
Silencing errors is not a fix, but just a workaround and a recipe for a disaster.

**Comment by bozhodimitrov** [comment]
Yes, I understand your point, but the underlying fundamental problem here is that there is non UTF-8 encoding errors, which are masked as the replacement character instead of fixing the actual issue, which is to use the proper decoding codec, which will result in no errors.

Your solution is partly acceptable, because it allows the user to get a better error message from the third-party layer tesseract.
But I am pretty sure it will break current expected logic, because the behavior changes, no?

Does the `OCR failed ...` message comes from a raised `TesseractError` exception?

**Comment by bozhodimitrov** [comment]
Can you provide an example of such broken input for the purposes of unit testing?
And when you report issues consider this approach: [The most important skill I've learned for software engineering (beginner) anthony explains [#340](https://www.youtube.com/hashtag/340)](https://www.youtube.com/watch?v=ritp4gAqNMI)

Not trying to teach you, just wanted to share that such info will make it a lot easier to accept such requests.

**Comment by bozhodimitrov** [comment]
The temp dir used by pytesseract is just the builtin option chosen by the standard library of Python itself.
So this is not a concern. And the user can modify that as you pointed. So if we want a failure test case, we need to reproduce the error. Otherwise there is no point of reporting such issue.
